module/documents/chat-message.mjs

```diff
@@ -307,7 +307,11 @@ export default class ChatMessage5e extends ChatMessage {
     if ( this.isContentVisible && item && roll ) {
       const isCritical = (roll.type === "damage") && this.rolls[0]?.isCritical;
       const subtitle = roll.type === "damage"
-        ? isCritical ? game.i18n.localize("DND5E.CriticalHit") : game.i18n.localize("DND5E.DamageRoll")
+        ? isCritical
+          ? game.i18n.localize("DND5E.CriticalHit") 
+          : activity.type === "heal"
+            ? game.i18n.localize("DND5E.HealingRoll")
+            : game.i18n.localize("DND5E.DamageRoll")
```

Comment:
`activity` isn't guaranteed to exist here. We can also use `damageFlavor` here after adjusting the flavor for `HealActivity`:
```suggestion
          ? game.i18n.localize("DND5E.CriticalHit")
          : activity?.damageFlavor ?? game.i18n.localize("DND5E.DamageRoll")
```